ext/telemetry/telemetry.ts

```diff
@@ -157,482 +145,320 @@ interface IKeyValue {
   key: string;
   value: IAnyValue;
 }
-interface IResource {
-  attributes: IKeyValue[];
-  droppedAttributesCount: number;
-}
 
 interface InstrumentationLibrary {
   readonly name: string;
   readonly version?: string;
   readonly schemaUrl?: string;
 }
 
-interface ReadableSpan {
-  readonly name: string;
-  readonly kind: SpanKind;
-  readonly spanContext: () => SpanContext;
-  readonly parentSpanId?: string;
-  readonly startTime: HrTime;
-  readonly endTime: HrTime;
-  readonly status: SpanStatus;
-  readonly attributes: SpanAttributes;
-  readonly links: Link[];
-  readonly events: TimedEvent[];
-  readonly duration: HrTime;
-  readonly ended: boolean;
-  readonly resource: IResource;
-  readonly instrumentationLibrary: InstrumentationLibrary;
-  readonly droppedAttributesCount: number;
-  readonly droppedEventsCount: number;
-  readonly droppedLinksCount: number;
+function hrToSecs(hr: [number, number]): number {
+  return (hr[0] * 1e3 + hr[1] / 1e6) / 1000;
 }
 
-enum ExportResultCode {
-  SUCCESS = 0,
-  FAILED = 1,
+export function enterSpan(span: Span): Context | undefined {
+  if (!span.isRecording()) return undefined;
+  const context = (CURRENT.get() || ROOT_CONTEXT).setValue(SPAN_KEY, span);
+  return CURRENT.enter(context);
 }
 
-interface ExportResult {
-  code: ExportResultCode;
-  error?: Error;
+export function restoreContext(context: Context): void {
+  setAsyncContext(context);
 }
 
-function hrToSecs(hr: [number, number]): number {
-  return ((hr[0] * 1e3 + hr[1] / 1e6) / 1000);
+function isDate(value: unknown): value is Date {
+  return ObjectPrototypeIsPrototypeOf(value, DatePrototype);
 }
 
-const TRACE_FLAG_SAMPLED = 1 << 0;
+interface OtelTracer {
+  __key: "tracer";
 
-const instrumentationScopes = new SafeWeakMap<
-  InstrumentationLibrary,
-  { __key: "instrumentation-library" }
->();
-let activeInstrumentationLibrary: WeakRef<InstrumentationLibrary> | null = null;
+  // deno-lint-ignore no-misused-new
+  new (name: string, version?: string, schemaUrl?: string): OtelTracer;
```

Comment:
I believe this is a false-positive from deno lint? `new()` should be allowed in interfaces

Reply:
Agreed